blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BlueTable.java

```diff
@@ -0,0 +1,32 @@
+package io.jenkins.blueocean.rest.model;
+
+import com.google.common.annotations.Beta;
+import org.kohsuke.stapler.export.Exported;
+import org.kohsuke.stapler.export.ExportedBean;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A Table-like data structure for serving tabular data
+ */
+@ExportedBean
+@Beta
+public abstract class BlueTable {
+
+    public static final String COLUMNS = "columns";
+    public static final String ROWS = "rows";
+
+    @Exported(name = COLUMNS, inline = true)
+    public abstract Map<String, String> getColumns();
+
+    @Exported(name = ROWS)
+    public abstract List<Row> getRows();
+
+    @ExportedBean(defaultVisibility = 2)
+    public abstract static class Row {
```

Comment:
Should be moved in to its own file unless its use is local.

Reply:
Docs

Reply:
Refactored out and docs added in 7b25272